lib/matplotlib/axes/_axes.py

```diff
@@ -6281,24 +6281,19 @@ def _normalize_input(inp, ename='input'):
                     xvals.append(x.copy())
                     yvals.append(y.copy())
 
-            if fill:
-                # add patches in reverse order so that when stacking,
-                # items lower in the stack are plottted on top of
-                # items higher in the stack
-                for x, y, c in reversed(list(zip(xvals, yvals, color))):
-                    patches.append(self.fill(
-                        x, y,
-                        closed=True,
-                        facecolor=c,
-                        margins=margins))
-            else:
-                for x, y, c in reversed(list(zip(xvals, yvals, color))):
-                    split = 2 * len(bins)
-                    patches.append(self.fill(
-                        x[:split], y[:split],
-                        closed=False, edgecolor=c,
-                        fill=False,
-                        margins=margins))
+            #stepfill is closed, step is not
+            split = -1 if fill else 2 * len(bins)
+            # add patches in reverse order so that when stacking,
+            # items lower in the stack are plottted on top of
+            # items higher in the stack
+            for x, y, c in reversed(list(zip(xvals, yvals, color))):
+                patches.append(self.fill(
+                    x[:split], y[:split],
+                    closed=True if fill else None,
+                    facecolor=c,
```

Comment:
No, because otherwise `histype=stepfilled` would have colored edges instead of black ones:

`filled=None` != `filled=False`


<img width="699" alt="screen shot 2016-05-29 at 13 59 13" src="https://cloud.githubusercontent.com/assets/335567/15635980/9058bebe-25a5-11e6-94f4-a85368363b20.png">

But, I thought so too in the beginning. 

(Compare above screenshot with test file in the PR)